src/main/groovy/com/netgrif/application/engine/petrinet/domain/dataset/logic/action/ActionDelegate.groovy

```diff
@@ -218,39 +227,121 @@ class ActionDelegate {
     }
 
     /**
-     * Changes behavior of a given field on given transition if certain condition is being met.
+     * Changes behavior of a given field on given transition if certain condition is being met or on each transition containing a field.
      * <br>
-     * Example:
+     * Example 1:
      * <pre>
      *     condition: f.conditionId,
      *     text: f.textId,
      *     transition: t.transitionId;
      *
      *     make text,visible on transition when { condition.value == true }* </pre>
-     * This code will change the field <i>text</i> behaviour to <i>visible</i> when fields <i>condition</i> value is equal to <i>true</i>
+     * This code will change the field <i>text</i> behaviour to <i>visible</i> when fields <i>condition</i> value is equal to <i>true</i>.
+     *
+     * Example 2:
+     * <pre>
+     *      text: f.textId;
+     *
+     *      if (condition)
+     *          make text, visible on each
+     * </pre>
+     * This code will change the field <i>text</i> behaviour to <i>visible</i> on each transition that contains the field <i>text</i>.
      * @param field which behaviour will be changed
-     * @param behavior one of visible, editable, required, optional, hidden, forbidden
+     * @param behavior one of initial, visible, editable, required, optional, hidden, forbidden
      */
     def make(Field field, Closure behavior) {
-        [on: { Transition trans ->
-            [when: { Closure condition ->
-                if (condition()) {
-                    behavior(field, trans)
-                    ChangedField changedField = new ChangedField(field.stringId)
-                    changedField.addAttribute("type", field.type.name)
-                    changedField.addBehavior(useCase.dataSet.get(field.stringId).behavior)
-                    SetDataEventOutcome outcome = createSetDataEventOutcome()
-                    outcome.addChangedField(field.stringId, changedField)
-                    this.outcomes.add(outcome)
+        [on: { Object optionalTransition ->
+            if (optionalTransition instanceof Transition) {
+                [when: { Closure condition ->
+                    if (condition()) {
+                        behavior(field, optionalTransition)
+                        saveFieldBehavior(field, optionalTransition)
+                    }
+                }]
+            } else if (optionalTransition instanceof Closure) {
+                if (optionalTransition == each) {
```

Comment:
It should throw an error or warning if provided closure is not one of the expected values.

Reply:
unexpected values will throw exception